MBBSEmu/CPU/CPUCore.cs

```diff
@@ -3536,43 +3509,24 @@ private void Op_Fdivrp()
             }
         }
 
-        [MethodImpl(CompilerOptimizations)]
-        public void Flags_EvaluateZero(ushort result = 0)
+        /// <summary>
+        ///     Evaluates and sets the value of both the Carry and Zero Flag based upon the specified result
+        /// </summary>
+        /// <param name="result"></param>
+        private void Flags_EvaluateSignZero(byte result)
         {
-            if (result == 0)
-            {
-                Registers.F = Registers.F.SetFlag((ushort)EnumFlags.ZF);
-            }
-            else
-            {
-                Registers.F = Registers.F.ClearFlag((ushort)EnumFlags.ZF);
-            }
+            Registers.F = result.IsNegative() ? Registers.F.SetFlag((ushort)EnumFlags.SF) : Registers.F.ClearFlag((ushort)EnumFlags.SF);
```

Comment:
if result.IsNegative() you can skip evaluation of Zero since it's obviously not 0

e.g.

```
if (result.IsNegative()) {
  setFlag(SF);
  clearFlag(ZF);
} else {
  clearFlag(SF);
  if (result == 0) .... set | clear
}
```